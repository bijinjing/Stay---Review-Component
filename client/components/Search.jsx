import React from 'react'

class Search extends React.Component{
  constructor(props) {
    super(props)

    this.changeHandler = this.changeHandler.bind(this);
    this.submitHandler = this.submitHandler.bind(this);
  }

  changeHandler({target}){
    this.props.changeSearchValue(target.value)
  }

  submitHandler(event){
    event.preventDefault()
    this.props.submitSearchValue()
  }

  render(){
    return(
        <form onSubmit={this.submitHandler}>
          <input
            id="search_text"
            type="text"
            value={this.props.searchValue}
            placeholder="Search reviews"
            onChange={this.changeHandler}
          />
        </form>
    )
  }
}

export default Search