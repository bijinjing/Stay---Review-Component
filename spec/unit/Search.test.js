import React from 'react';
import Search from '../../client/components/Search.jsx';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import {mount} from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

describe('Search', ()=>{


  it('renders without crashing', () => {
    const props = {
      searchValue:"hello"
    }

    const wrapper = shallow(<Search {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should run changeHandler with changed text in search box', () => {
    let result = null;
    const props = {
      changeSearchValue:()=>{result="hit"}
    }
    const component = mount(<Search {...props}/>);
    component
      .find('#search_text')
      .simulate('change');

    expect(result).toEqual("hit");
    component.unmount();
  });

  it('should run submitHandler with submitting a search text in search box', () => {
    let result = null;
    const props = {
      submitSearchValue:()=>{result="hit"}
    }
    const component = mount(<Search {...props}/>);
    component
      .find('form')
      .simulate('submit');

    expect(result).toEqual("hit");
    component.unmount();
  });
})