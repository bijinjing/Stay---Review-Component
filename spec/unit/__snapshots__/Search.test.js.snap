// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders without crashing 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    id="search_text"
    onChange={[Function]}
    placeholder="Search reviews"
    type="text"
    value="hello"
  />
</form>
`;
